### PR TITLE
fix: include hidden passkey fields in form

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: src/react-components/ory/user-auth-card.tsx [hunk 1]
```diff
@@ -114,31 +115,33 @@ export type UserAuthCardProps = {
...
+    useScriptNodes({ nodes: flowProps.flow.ui.nodes })
   }
 
   if (!title) {
-    switch (flowType) {
+    switch (flowProps.flowType) {
       case "login":
-        if (flow.refresh) {
+        if (flowProps.flow.refresh) {
           title = intl.formatMessage({
```
Comment (jonas-jonas) [inline src/react-components/ory/user-auth-card.tsx:140]: Type narrowing doesn't work on individual variables, that determine the type of the other. You need them to be part of an object. Then you can assert the type of one of its properties to narrow the whole type down.

File: src/react-components/ory/helpers/user-auth-form.tsx [hunk 2]
```diff
@@ -108,23 +98,37 @@ export const UserAuthForm = ({
...
+      onSubmit={submitHandler}
+      {...props}
+    >
+      <>
+        {/*always add csrf token and other hidden fields to form*/}
+        <FilterFlowNodes
+          filter={
+            formFilterOverride ?? {
+              nodes: flow.ui.nodes,
+              groups: ["default", "passkey"], // we only want to map hidden default fields here
```
Comment (jonas-jonas) [inline src/react-components/ory/helpers/user-auth-form.tsx:124]: passkey change.

File: src/react-components/ory/user-auth-card.tsx (hunk 1 above)
Comment (aeneasr) [reply src/react-components/ory/user-auth-card.tsx:140]: I think you need to explain that to me like I‘m 5 😅

Comment (aeneasr) [reply src/react-components/ory/user-auth-card.tsx:140]: Oh I get it

**Pull request merged**